erpnext/public/js/utils.js

```diff
@@ -590,6 +591,20 @@ erpnext.utils.map_current_doc = function(opts) {
 	}
 }
 
+erpnext.utils.clear_duplicates = function() {
+	const unique_items = new Map();
+	let items = []
+
+	for (let i = 0; i < cur_frm.doc.items.length; i++) {
+		let item = cur_frm.doc.items[i];
+		if (!(unique_items.has(item.item_code) && unique_items.get(item.item_code) === item.qty)) {
```

Comment:
This will misbehave if you are delivering from different warehouses, or have two different batches for an item with the same quantity.